### PR TITLE
Allow skipping preflights

### PR DESCRIPTION
**Pull request by divolgin**
On initial install, if a license and a config files are provided, skip-preflights will bypass preflight checks and deploy the app.
```
kots install myapp --config-values ./config.yaml --license-file ./license.yaml --shared-password password --skip-preflights
```

On updates, skip-preflights will bypass prefligths.  To deploy the latest version, the `--deploy` flag should be specified.
```
kots upstream upgrade myapp -n myapp --skip-preflights
```

### CODE REVIEW

**Comment by markpundsack** [comment]
@divolgin Can you add the description of the new option to the PR?

File: kotsadm/pkg/store/ocistore/version_store.go [hunk 1]
```diff
@@ -331,49 +331,46 @@ func (s OCIStore) CreateAppVersion(appID string, currentSequence *int64, appName
...
-			}
-			if t {
-				downstreamStatus = "pending_config"
-			}
+		downstreamStatus := "pending"
+		if currentSequence == nil && needsConfig {
+			downstreamStatus = "pending_config"
+		} else if kotsKinds.Preflight != nil && !skipPreflights {
+			downstreamStatus = "pending_preflight"
+		} else if currentSequence != nil && needsConfig {
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/ocistore/version_store.go:372]: i think this should come before the preflight condition. probably the first if statement should just check for `needsConfig`.

Comment (divolgin) [reply kotsadm/pkg/store/ocistore/version_store.go:372]: In the original logic it looks like `pending_config` overrides `pending_preflight`.  So I kept that.

File: kotsadm/pkg/store/s3pg/version_store.go [hunk 2]
```diff
@@ -318,49 +318,46 @@ func (s S3PGStore) CreateAppVersion(appID string, currentSequence *int64, appNam
...
-			}
-			if t {
-				downstreamStatus = "pending_config"
-			}
+		downstreamStatus := "pending"
+		if currentSequence == nil && needsConfig {
+			downstreamStatus = "pending_config"
+		} else if kotsKinds.Preflight != nil && !skipPreflights {
+			downstreamStatus = "pending_preflight"
+		} else if currentSequence != nil && needsConfig {
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/s3pg/version_store.go:359]: same here

**Review by salahalsaleh — APPROVED** [review]

**Pull request merged**